pages/k8s/troubleshooting.md

```diff
@@ -209,3 +209,197 @@ This is caused by the API load balancer not forwarding ports in the context of t
 ## Logging and monitoring
 
 By default there is no log aggregation of the Kubernetes nodes, each node logs locally. Please read over the [logging](../logging) page for more information.
+
+## Troubleshooting Keystone/LDAP issues
+
+The following section offers some notes to help determine issues with using Keystone
+for authentication/authorisation.
+
+Testing the steps is important to determine the cause of the problem.
+
+### Can you communicate with Keystone and get an authorization token.
+
+First is to verify that Keystone communication works from both your client and
+the kubernetes-worker machines. The easiest thing to do here is to copy the
+kube-keystone.sh script to the machines of interest from kubernetes-master with
+`juju scp kubernetes-master/0:kube-keystone.sh .`, edit the script to include
+you credentials, `source kube-keystone.sh` and then run `get_keystone_token`.
+This will produce a token from the Keystone server. If that isn't working,
+check firewalls settings and your Keystone server. Note that the
```

Comment:
should this be "firewall settings on your Keystone server"?

Reply:
It is both your client and Keystone server really, but I think what we ended up on will be fine.